include/watchdog.h

```diff
@@ -28,19 +28,6 @@
 extern "C" {
 #endif
 
-/**
- * @name Deprecated Watchdog API types.
- * @{
- */
-#define WDT_MODE		(BIT(1))
-#define WDT_MODE_OFFSET		(1)
-#define WDT_TIMEOUT_MASK	(0xF)
-
-enum wdt_mode {
-	WDT_MODE_RESET = 0,
-	WDT_MODE_INTERRUPT_RESET
```

Comment:
`WDT_MODE_RESET` and `WDT_MODE_INTERRUPT_RESET` are being used in files _drivers/watchdog/wdt_esp32.c_and _./drivers/watchdog/wdt_sam0.c_, consider removing from them too.

Reply:
moved wdt_mode into esp32.c the code in wdt_sam0.c isn't using that mode enum.